main.rb

```diff
@@ -0,0 +1,199 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr = to_a if instance_of?(Range)
+    arr = flatten if instance_of?(Hash)
+
+    counter = 0
+    while counter < arr.length
+      yield(arr[counter])
+      counter += 1
+    end
+    arr
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr = to_a if instance_of?(Range)
+    arr = flatten if instance_of?(Hash)
+    counter = 0
+    while counter < arr.length
+      yield(arr[counter], counter)
+      counter += 1
+    end
+    arr
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each { |value| new_arr << value if yield(value) }
+    new_arr
+  end
+
+  def my_all?(args = nil)
+    if block_given?
+      my_each { |value| return false if yield(value) == false }
+      return true
+    elsif args.nil?
+      my_each { |v| return false if v.nil? || v == false }
+    elsif !args.nil? && (args.is_a? Class)
+      my_each { |v| return false if v.class != args }
+    elsif !args.nil? && args.instance_of?(Regexp)
+      my_each { |v| return false unless args.match(v) }
+    else
+      my_each { |v| return false if v != args }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |value| return true if yield(value) }
+      false
+    elsif arg.nil?
+      my_each { |v| return true if v.nil? || v == true }
+    elsif !arg.nil? && (arg.is_a? Class)
+      my_each { |v| return true if v.instance_of?(arg) }
+    elsif !arg.nil? && arg.instance_of?(Regexp)
+      my_each { |v| return true if arg.match(v) }
+    else
+      my_each { |v| return true if v == arg }
+    end
+    false
+  end
```

Comment:
- my_any? when no block or argument is given returns false if at least one of the collection is not true
- my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric